packages/salesforce-adapter/src/filters/standard_value_sets.ts

```diff
@@ -196,55 +215,88 @@ const emptyFileProperties = (fullName: string): FileProperties => ({
-          .filter(isInstanceElement)
-          .map(async inst => {
-            const clone = inst.clone()
-            await resolveTypeShallow(clone, config.elementsSource)
-            return clone
-          })
-          .filter(isInstanceOfType(STANDARD_VALUE_SET))
-          .toArray()
-      await updateSVSReferences(customObjectTypeElements, svsInstances)
-    }
-
-    return {
-      configSuggestions: configChanges,
-    }
-  },
-})
+): RemoteFilterCreator => ({ client, config }) => {
+  let originalChanges: Record<string, Change>
+  return {
+    /**
+     * Upon fetch, retrieve standard value sets and
+     * modify references to them in fetched elements
+     *
+     * @param elements the already fetched elements
+     */
+    onFetch: async (elements: Element[]): Promise<FilterResult> => {
+      const svsMetadataType: ObjectType | undefined = await findStandardValueSetType(elements)
+      let configChanges: ConfigChangeSuggestion[] = []
+      let fetchedSVSInstances: InstanceElement[] | undefined
+      if (svsMetadataType !== undefined) {
+        const svsInstances = await fetchMetadataInstances({
+          client,
+          fileProps: [...standardValueSetNames].map(emptyFileProperties),
+          metadataType: svsMetadataType,
+          metadataQuery: config.fetchProfile.metadataQuery,
+        })
+        elements.push(...svsInstances.elements)
+
+        configChanges = svsInstances.configChanges
+        fetchedSVSInstances = svsInstances.elements
+      }
+
+      const customObjectTypeElements = await awu(elements)
+        .filter(isObjectType)
+        .filter(isCustomObject)
+        .toArray()
+
+      if (customObjectTypeElements.length > 0) {
+        const svsInstances = fetchedSVSInstances !== undefined
+          ? fetchedSVSInstances
+          : await awu(await config.elementsSource.getAll())
+            .filter(isInstanceElement)
+            .map(async inst => {
+              const clone = inst.clone()
+              await resolveTypeShallow(clone, config.elementsSource)
+              return clone
+            })
+            .filter(isInstanceOfType(STANDARD_VALUE_SET))
+            .toArray()
+        await updateSVSReferences(customObjectTypeElements, svsInstances)
+      }
+
+      return {
+        configSuggestions: configChanges,
+      }
+    },
+    preDeploy: async changes => {
+      const standardPicklistFieldChanges = await awu(changes)
+        .filter(isModificationChange)
+        .filter(isFieldChange)
+        .filter(change => isStandardPickList(getChangeData(change)))
+        .toArray()
+      originalChanges = await keyByAsync(
+        standardPicklistFieldChanges,
+        change => apiName(getChangeData(change))
+      )
+      const deployableChanges = standardPicklistFieldChanges
+        .map(toDeployableStandardPicklistFieldChange)
+      _.pullAll(changes, standardPicklistFieldChanges)
+      changes.push(...deployableChanges)
```

Comment:
should not use `push(...` as it can crash with sufficiently large lists (it tries to unpack the list on the stack and if the list is long enough, it overflows and crashes)
replace with
```suggestion
      deployableChanges.forEach(change =>  changes.push(change))
```